IdentityCore/src/cache/accessor/MSIDAccountCredentialCache.m

```diff
@@ -306,6 +306,7 @@ - (BOOL)removeCredential:(nonnull MSIDCredentialCacheItem *)credential
                                                                                             clientId:credential.clientId
                                                                                       credentialType:credential.credentialType];
 
+    key.familyId = credential.familyId;
     key.realm = credential.realm;
     key.target = credential.target;
```

Comment:
As a reference, the branch consumed by MSAL doesn't have such an issue:
https://github.com/AzureAD/microsoft-authentication-library-common-for-objc/blob/dev/IdentityCore/src/cache/accessor/MSIDAccountCredentialCache.m#L330